leader/leader.go

```diff
@@ -126,15 +132,17 @@ func (l *Client) getWatchAtLatestIndex(ctx context.Context, api client.KeysAPI,
 // AddWatch starts watching the key for changes and sending them
 // to the valuesC, the watch is stopped
 func (l *Client) AddWatch(key string, retry time.Duration, valuesC chan string) {
-	prefix := fmt.Sprintf("AddWatch(key=%v)", key)
 	api := client.NewKeysAPI(l.client)
 
+	logger := log.WithField("key", key)
+	logger.Info("Setting up watch.")
```

Comment:
I would consider adding some fields for the configuration or logging as separate lines all the configuration, specifically the retry duration and etcd peers. My style for a log at entry like this, is not so much to log that this has started, which could just as easily be seen from the logs in getWatchAtLatest index, but to provide the context about what's been configured. I'm assuming here that I remember correctly how AddWatch is used, where it's only really called once during setup or if it has to be reinitialized for some reason.